docs/Customization/Custom-Core.rst

```diff
@@ -0,0 +1,267 @@
+.. _custom_core:
+
+Adding a custom core
+====================
+
+You may want to integrate a custom RISC-V core into the Chipyard framework. This documentation page provides a step-to-step
+instruction on how to achieve this.
+
+.. note:: 
+
+    RoCC is currently not supported by cores other than Rocket and BOOM. Please use Rocket or BOOM as the RoCC base core if you need to use RoCC.
+
+
+Wrap Verilog Module with Blackbox (Optional)
+--------------------------------------------
+
+Since Chipyard uses Scala and Chisel, if the top module of your core is not in Chisel, you will first need to create a Verilog
+blackbox for it so that it can be processed by Chipyard. See :ref:`incorporating-verilog-blocks` for instructions.
+
+Create Parameter Case Classes
+-----------------------------
+
+Chipyard will generate a core for every ``InstantiableTileParams`` object it discovered in the ``TilesLocated(InSubsystem)`` key.
+This object is derived from``TileParams``, a trait containing the information needed to create a tile. All cores must have
+their own implementation of ``InstantiableTileParams``, as well as ``CoreParams`` which is passed as a field in ``TileParams``.
+
+``TileParams`` holds the parameters for the tile, which are the same for every generated core, while ``CoreParams`` 
+contains the parameters for individual cores. They must be implemented as case classes with fields that can be overridden by 
+other config fragments as the constructor parameters. See the appendix at the bottom of the page for a list of 
+variable to be implemented. You can also add custom fields to them, but standard fields should always be preferred. 
+
+``InstantiableTileParams[TileType]`` holds the constructor of ``TileType`` on top of the fields of ``TileParams``.
+All custom cores will also need to implement ``instantiate()`` in their tile parameter class to return a new instance
+of the tile class ``TileType``. 
+
+``TileParams``, ``InstantiableTileParams[TileType]`` and ``CoreParams`` contains the following fields (you may ignore 
+any fields marked "Rocket specific" and use their default values, although it is recommended to use them if you 
```

Comment:
Does this mean that any field marked as "Rocket specific" does not need to be implemented by a custom core?

Reply:
Yes, and I mentioned above the code section. I moved that note down here in this revision to make it easier to understand.